### PR TITLE
refactor(makeFlat): use internal makeFlat

### LINKED ISSUE #1364: flattenDepth
**Is your feature request related to a problem? Please describe.**
`flattenDepth` function contains reference to internal ramda function called `_makeFlat` which further makes reference to `_isArrayLike`. We have `_isArrayLike` function implemented in ramda-adjunct, so we only need to reimplement `makeFlat` into our codebase. Be aware that function just cannot be copied due to licensing reasons, it really needs to be understood and reimplemented. 

**Describe the solution you'd like**
See text above..

**Describe alternatives you've considered**

--

**Additional context**

--

### PR DESCRIPTION
**Pull request by ryanbas21**
This addresses the issue #1364 which asks to use an internal version of makeFlat for flattenDepth
rather than the ramda one.

This is essentially the ramda one with some changes for the linting and closes #1364

### CODE REVIEW

File: src/makeFlat.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import _isArrayLike from './isArrayLike';
```
Comment (char0n) [inline src/makeFlat.js:1]:
```suggestion
import isArrayLike from './isArrayLike';
```

File: src/makeFlat.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
```
Comment (char0n) [inline src/makeFlat.js:3]: Can we move this to `src/internal/makeFlat` please ?

File: test/makeFlat.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { assert } from 'chai';
+
+import * as RA from '../src';
+
+describe('makeFlat', function () {
```
Comment (char0n) [inline test/makeFlat.js:5]: Can we please move this to `test/internal/makeFlat.js`?

File: src/makeFlat.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * @memberOf RA
+ *
+ * @category List
+ * @param {!bool} = should recursively flatten
+ * @param {!Array} = the nested list to be flattened
+ * @return {!Array} = the flattened list
+ * @sig Bool -> List -> List
+ *
+ */
+export default function makeFlat(recursive) {
```
Comment (char0n) [inline src/makeFlat.js:17]: Let's make the function declaration more modern.
```js
const makeFlat = recursive => list  => {
  ...body of the function..
};
```

File: src/index.js [hunk 5]
```diff
@@ -154,6 +154,7 @@ export { default as skipTake } from './skipTake';
 export { default as paths } from './paths';
 export { default as renameKeys } from './renameKeys';
 export { default as renameKeysWith } from './renameKeysWith';
+export { default as makeFlat } from './makeFlat';
```
Comment (char0n) [inline src/index.js:157]: No need to export this as public API.

File: src/makeFlat.js (hunk 4 above)
Comment (ryanbas21) [reply src/makeFlat.js:17]: how do you suppose calling the recursion on an anonymous function?

File: test/internal/makeFlat.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { assert } from 'chai';
+
+import * as RA from '../../src';
+import makeFlat from '../../src/internal/makeFlat';
+
+describe('makeFlat', function () {
+  it('should flatten an ar of n levels', function () {
```
Comment (char0n) [inline test/internal/makeFlat.js:7]:
```suggestion
  it('should flatten a list of n levels', function () {
```

File: src/makeFlat.js (hunk 4 above)
Comment (char0n) [reply src/makeFlat.js]: Right, you're absolutely correct. Never mind my comment.

**Comment by char0n** [comment]
@ryanbas21 can you please return changes you did to `CONTRIBUTING.md` file back to original state. This is the only thing that is preventing me to merge this PR. Nicely done here and thank you for your contribution!

**Comment by ryanbas21** [comment]
> @ryanbas21 can you please return changes you did to `CONTRIBUTING.md` file back to original state. This is the only thing that is preventing me to merge this PR. Nicely done here and thank you for your contribution!

pushed. not sure how that happened, sorry about that

**Comment by char0n** [comment]
@ryanbas21 no problem. THank you for your work. Merging.

**Pull request merged**